leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -39,17 +38,526 @@ var EvolvingFunctionBBM = Backbone.Model.extend({
+        var intentionEval = analysisRequest.getUserEvaluationByID(this.get('id'), '0');
+        intentionEval.evaluationValue = initValue;
+ 
+        // if there is only one function segment, and its constant, then we need to
+        // change the function segment's marked value
+ 
+        var funcSegList = this.evolvingFunction.get('functionSegList');
+        
+        if (this.evolvingFunction.get('stringDynVis') == 'C' || 
+            (this.evolvingFunction.get('stringDynVis') == 'UD' && funcSegList[0].get('type') == 'C')) { 
+                functionSegList[0].set('refEvidencePair', initValue); 
+            }
+        this.evolvingFunction.set('stringDynVis', 'NT');
+        this.evolvingFunction.set('functionSegList', []);
+    }, 
+ 
+    /**
+     * Sets the initial satisfaction value for this Intention to '(no value)'
+     */
+    removeInitialSatValue: function() {
+        this.changeInitialSatValue('(no value)');
+    },    
+
+    /**
+     * Sets newID as the new ID for this intention
+     */
+    // TODO - update this function
+    // if we are using the automatically generated 'id', then we can just call 
+    // it whenever we need?? and would we be not allowed to change it??
+    setNewID: function(newID) {
+        var oldID = this.get('id');   
+        this.set('id', newID); 
+        this.evolvingFunction.set('id', newID); 
+ 
+        // UserAssignementsList
+        var userAssignList = analysisRequest.userAssignmentsList;
+        for (var i = 0; i < userAssignList.length; i++) {
+            if (userAssignList[i].intentionID === oldID) {
+                userAssignList[i].intentionID = newID;
+            }
+        }
+ 
+        // Links
+        var links = model.links;
+        for (var i = 0; i < links.length; i++) {
+            if (links[i].linkSrcID === oldID) {
+                links[i].linkSrcID = newID;
+            }
+            if (links[i].linkDestID === oldID) {
+                links[i].linkDestID = newID;
+            }
+        }
+        
+        // Constraints
+        var consts = model.constraints;
+        for (var i = 0; i < consts.length; i++) {
+            if (consts[i].constraintSrcID === oldID) {
+                consts[i].constraintSrcID = newID;
+            }
+            if (consts[i].constraintDestID === oldID) {
+                consts[i].constraintDestID = newID;
+            }
+        }
+ 
+    }, 
+ 
+    /**
+     * Returns the 4 digit representation for this
+     * Intention's initial satisfaction value
+     *
+     * @returns {String}
+     */
+    getInitialSatValue: function() {
+        var intentionEval = analysisRequest.getUserEvaluationByID(this.get('id'), '0');
+        if (typeof intentionEval == 'undefined'){
+            return '(no value)';
+        } else {
+            return intentionEval.evaluationValue;
+        }
+    }, 
+ 
+    /**
+     * Resets dynamic function
+     */
+    removeFunction: function() {
+        this.removeAbsCosnt();
```

Comment:
Does cosnt mean constraint? It looks like const(ant) spelled incorrectly so maybe we could rename?